src/controllers/contract/index.ts

```diff
@@ -0,0 +1,98 @@
+import { Request, Response } from 'express';
+import ContractService from '../../services/contract';
+import { CreateContractRequest, FindContractRequest, PatchContractRequest } from './types';
+import { handlePrismaError } from '../../lib/handle-prisma-error';
+import UserService from '../../services/user';
+import ContractCategoryService from '../../services/contractCategories';
+
+export default class ContractController {
+  static async getAll(_: Request, response: Response) {
+    const contracts = await ContractService.getAll();
+    return response.success({ data: contracts });
+  }
+
+  static async find(request: Request, response: Response) {
+    const { id } = request.params as any as FindContractRequest;
+
+    const contract = await ContractService.find({ id });
+
+    if (!contract) {
+      return response.notfound();
+    }
+
+    return response.success({ data: contract });
+  }
+
+  static async create(request: Request, response: Response) {
+    const body = request.body as any as CreateContractRequest;
+
+    try {
+      const beneficiary = await UserService.find({ id: body.beneficiary_id });
+      const provider = await UserService.find({ id: body.provider_id });
+      const category = await ContractCategoryService.find({ id: body.category_id });
+
+      if (!beneficiary) {
+        // eslint-disable-next-line prettier/prettier
+        return response.badrequest({ errors: { beneficiary_id: 'Invalid beneficiary ID provided' } });
+      }
+      if (!provider) {
+        return response.badrequest({ errors: { provider_id: 'Invalid provider ID provided' } });
+      }
+      if (!category) {
+        return response.badrequest({ errors: { category_id: 'Invalid category ID provided' } });
+      }
+    } catch (err) {
+      return response.error(handlePrismaError(err));
+    }
+
+    if (!body.beneficiary_id) {
+      return response.badrequest({ errors: { beneficiary_id: 'Beneficiary is required ' } });
+    }
+    if (!body.provider_id) {
+      return response.badrequest({ errors: { provider_id: 'Provider is required' } });
+    }
+    if (!body.category_id) {
+      return response.badrequest({ errors: { category_id: 'Category is required' } });
+    }
```

Comment:
how come you are checking these after you try to fetch from the database.

this should be before fetching the database to check if it exists

Reply:
My mistake, changed to before